src/ApiKitClient.ts

```diff
@@ -28,8 +36,11 @@ export class ApiKitClient {
     });
 
     this.instance.interceptors.response.use((response: AxiosResponse) => response, (error: AxiosError) => {
-      if (error.response?.status === 401 && this.unauthorizationCallback) {
+      const { status } = error.response;
+      if (status === 401 && this.unauthorizationCallback) {
         this.unauthorizationCallback();
+      } else if (this.globalErrorCallback) {
+        this.globalErrorCallback();
```

Comment:
Like this
```suggestion
        this.globalErrorCallback(error);
```

Reply:
This is a callback doesn't accept any parameter why should I pass the error object to it?!

Reply:
@MostafaFawzy7 to let the consumer know what exactly happened. Give the consuming app some context. This is actually a pretty good idea.

Reply:
@MostafaFawzy7 to let the consumer know what exactly happened. Give the consuming app some context. This is actually a pretty good idea.

Reply:
@HassanElZarkawy The idea of this callback is to add a custom error by the consumer at the global scope of error handling, why to pass it the `error` object which is already exposable by the `Promise` rejection at the next line of this condition?